build/scripts/DevEnv/NewDevEnv.ps1

```diff
@@ -0,0 +1,234 @@
+}
+
+function BuildApp {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $appProjectFolder,
+
+        [Parameter(Mandatory = $true)]
+        [string] $packageCacheFolder
+    )
+
+    $appProjectFolder = ResolveFolder -folder $appProjectFolder
+
+    $appInfo = [AppProjectInfo]::Get($appProjectFolder)
+    $appFile = $appInfo.GetAppFileName()
+
+    if(Test-Path (Join-Path $packageCacheFolder $appFile)) {
+        Write-Host "App $appFile already exists in $packageCacheFolder. Skipping..."
+        return (Join-Path $packageCacheFolder $appFile -Resolve)
+    }
+
+    $allAppInfos = GetAllApps
+
+    # Build dependencies
+    foreach($dependency in $appInfo.AppJson.dependencies) {
+        $dependencyAppInfo = $allAppInfos | Where-Object { $_.Id -eq $dependency.id }
+        $dependencyAppFile = BuildApp -appProjectFolder $dependencyAppInfo.AppProjectFolder -packageCacheFolder $packageCacheFolder
+        PublishApp -appFile $dependencyAppFile
+    }
+
+    $compilerFolder = CreateCompilerFolder -packageCacheFolder $packageCacheFolder
+
+    try {
+        $appFile = Compile-AppWithBcCompilerFolder -compilerFolder $compilerFolder -appProjectFolder "$($appInfo.AppProjectFolder)" -appOutputFolder $packageCacheFolder -CopyAppToSymbolsFolder -appSymbolsFolder $packageCacheFolder
+    }
+    finally {
+        Remove-Item -Path $compilerFolder -Recurse -Force | Out-Null
+    }
+
+    return $appFile
+}
+
+function PublishApp {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $appFile
+    )
+
+    Publish-BcContainerApp -containerName $script:containerName -appFile $appFile -syncMode ForceSync -sync -credential $script:credential -skipVerification -install -ignoreIfAppExists
+}
+
+$errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+Import-Module "$PSScriptRoot\..\EnlistmentHelperFunctions.psm1"
+
+$script:allApps = @()
+[pscredential] $script:credential = New-Object System.Management.Automation.PSCredential ($userName, $(ConvertTo-SecureString $password -AsPlainText -Force))
+$script:baseFolder = Get-BaseFolder
+$script:packageCacheFolder = ResolveFolder -folder $packageCacheFolder
+$script:containerName = $containerName
+
+$projectPaths = ResolveProjectPaths -projectPaths $projectPaths -workspacePath $workspacePath -alGoProject $alGoProject
+
+Write-Host "Loading BCContainerHelper module" -ForegroundColor Yellow
+InstallBCContainerHelper
+
+Write-Host "Creating container $script:containerName" -ForegroundColor Yellow
+CreateBCContainer
+
+foreach($currentProjectPath in $projectPaths) {
+    Write-Host "Building app in $currentProjectPath" -ForegroundColor Yellow
+    $appFile = BuildApp -appProjectFolder $currentProjectPath -packageCacheFolder $script:packageCacheFolder
+
+    Write-Host "Publishing app $appFile to $script:containerName" -ForegroundColor Yellow
+    PublishApp -appFile $appFile
+}
```

Comment:
I ran the script and got a BC environment with only the System Application installed. 
<img width="431" alt="image" src="https://github.com/microsoft/BCApps/assets/117829001/9ba046f1-15d3-4679-a339-b07c1ef0ce26">


I guess this is by design but is that what developers would expect? Wouldn't it be better to get a full bc environment with baseapp and first party apps + The system application from my branch?

Reply:
That's definitely a valid scenario! I'll see how it can be supported.

Reply:
Not going to be in this iteration. Future enhancement it is 😉